sdk/src/index.ts

```diff
@@ -6,13 +6,32 @@ import { DeviceBase, ScryptedInterfaceDescriptors, ScryptedInterfaceProperty, TY
  * @category Core Reference
  */
 export class ScryptedDeviceBase extends DeviceBase {
-  private _storage: Storage;
-  private _log: Logger;
-  private _console: Console;
-  private _deviceState: DeviceState;
+  private _storage: Storage | undefined;
+  private _log: Logger | undefined;
+  private _console: Console | undefined;
+  private _deviceState: DeviceState | undefined;
 
   constructor(public readonly nativeId?: string) {
     super();
+
+    addScryptedInterfaceProperties({
+      thisObject: this,
+      getSet: (field: keyof DeviceState) => {
+        return (value: any) => {
+          this._lazyLoadDeviceState();
+          if (!this._deviceState)
+            console.warn('device state is unavailable. the device must be discovered with deviceManager.onDeviceDiscovered or deviceManager.onDevicesChanged before the state can be set.');
+          else
+            this._deviceState[field] = value;
+        }
+      },
+      getGet: (field: keyof DeviceState) => {
+        return () => {
+          this._lazyLoadDeviceState();
+          return this._deviceState?.[field];
+        }
+      }
+    })
```

Comment:
I came up with this way to do this in a more type-safe manner, however it meant that we had to repeat it in both `class ScryptedDeviceBase` and `class MixinDeviceBase`